src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1572,7 +1572,7 @@ impl Cage {
                             // iterate over filename-inode pairs in dict
                             for (filename, inode) in dir_inode_obj.filename_to_inode_dict.clone().into_iter() {
                                 // convert filename to a filename vector of u8
-                                let mut vec_filename: Vec<u8> = filename.as_bytes().to_vec();
+                                let mut vec_filename: Vec<u8> = String::from(filename).as_bytes().to_vec();
```

Comment:
This needs to be an `interface::OsStringKey::from(filename)` as the String::from string is not null terminated, and we need to have a null terminated string to pass back to C.

Reply:
Ah I tried `interface::OsStringKey::from` and it does not exist. Did you mis-remember where it is located or its name?

Reply:
Ok the issue there is not that interface::OsStringKey::from doesn't exist, it's that it doesn't have an as_bytes method. So here's my fix:
Change that line to

`let mut vec_filename: Vec<u8> = interface::RustCStr::from_bytes_with_nul(filename.as_bytes()).unwrap().to_bytes().to_vec();`

And then change the line in interface/file.rs which is currently 

`pub use std::ffi::OsString as OsStringKey;`

to be 

`pub use std::ffi::CStr as RustCStr;`

Reply:
`from_bytes_with_nul()` requires the input slice to be null-terminated. Would it be easier if I just push NULL after I convert the string to the u8 vector?

Reply:
Yeah that's probably the simplest solution, and workable at least for now. No need to do anything more complicated than that.